satpy/dependency_tree.py

```diff
@@ -289,29 +289,25 @@ def _find_reader_node(self, dataset_key, query):
 
         """
         matching_ids = self._find_matching_ids_in_readers(dataset_key)
-        result = self._get_unique_matching_id(matching_ids, dataset_key, query)
+        unique_id = self._get_unique_matching_id(matching_ids, dataset_key, query)
 
         for reader_name, ids in matching_ids.items():
-            if result in ids:
-                data = {'reader_name': reader_name}
-                break
-        else:
-            raise RuntimeError("Data ID disappeared.")
+            if unique_id in ids:
+                return self._get_unique_reader_node_from_id(unique_id, reader_name)
 
-        result = self._get_unique_node_from_id(result, data)
-        return result
+        raise RuntimeError("Data ID disappeared.")
 
     def _find_matching_ids_in_readers(self, dataset_key):
         matching_ids = {}
-        for reader_name, reader_instance in self.readers.items():
-            matching_ids[reader_name] = []
+        for file_handler_name, reader_instance in self.readers.items():
```

Comment:
These aren't file handlers.

Reply:
Indeed. I got confused by looking at the values, forgetting that the *reader* name is also set in the yaml file for the format. Reverted.